packages/components/src/util/utils.ts

```diff
@@ -573,3 +573,47 @@ export function getUpdatedDataFromGrid(
 export const blurActiveElement = (): void => {
     (document.activeElement as HTMLElement).blur();
 };
+
+/**
+ * Unique names are not enforced for samples when they belong to different sample types, or data from different data classes.
+ * When generating SelectInput options, we want to append sample type or source type to entries with duplicate names so user know which item they are selecting.
+ * @param rows
+ * @param keyField
+ * @param valueField
+ * @param typeField
+ */
+export function getDisambiguatedSelectInputOptions(rows: List<any> | { [key: string] : any }, keyField: string, valueField: string, typeField: string) {
```

Comment:
Looks like the type of rows is really `List<Map<string, any>> | {[key:string] : any}`